chimney/src/main/scala/io/scalaland/chimney/internal/macros/TransformerMacros.scala

```diff
@@ -402,18 +402,29 @@ trait TransformerMacros extends TransformerConfiguration with MappingMacros with
             .getOrElse {
               val instSymbol = instTpe.typeSymbol
               targetNamedInstances.get(instName) match {
-                case Some(matchingTargetTpe)
+                case Some(List(matchingTargetTpe))
                     if (instSymbol.isModuleClass || instSymbol.isCaseClass) && matchingTargetTpe.typeSymbol.isModuleClass =>
                   val tree = mkTransformerBodyTree0(config) {
                     q"${matchingTargetTpe.typeSymbol.asClass.module}"
                   }
                   Right(cq"_: ${instSymbol.asType} => $tree")
-                case Some(matchingTargetTpe) if instSymbol.isCaseClass && matchingTargetTpe.typeSymbol.isCaseClass =>
+                case Some(List(matchingTargetTpe))
+                    if instSymbol.isCaseClass && matchingTargetTpe.typeSymbol.isCaseClass =>
                   val fn = freshTermName(instName)
                   expandDestinationCaseClass(Ident(fn), config.rec)(instTpe, matchingTargetTpe)
                     .mapRight { innerTransformerTree =>
                       cq"$fn: $instTpe => $innerTransformerTree"
                     }
+                case Some(list) if list.size > 1 =>
```

Comment:
Guess you need to use `scala.collection.compat` for 2.12 in order to use `sizeIs`. In case it's not available in compat,  `.sizeCompare(1) > 0` is your legacy alternative.

Reply:
Only for Scala 2.13. I think it's rare case when list size more than 1, and size computing optimization will not improve solution

Reply:
Replaced with pattern-matching, it should be faster